### PR TITLE
making scopus file converter more robust

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java [hunk 1]
```diff
@@ -167,10 +162,35 @@ private static Collector<ScopusFile, Object, ArrayList<ScopusFile>> collectRemov
...
+
+    private static URI toUri(String string) {
+        return attempt(() -> new URI(string.replace(WHITESPACE, ENCODED_WHITESPACE))).orElseThrow();
+    }
+
+    private static boolean fileWithContent(ScopusFile file) {
+        return file.size() != ZERO_LENGTH_CONTENT;
+    }
+
+    private static boolean isElsvierPlainTextResource(CrossrefLink crossrefLink) {
```
Comment (joachimjorgensen) [inline scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java:185]: Typo: isElsevier...()

File: scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java [hunk 2]
```diff
@@ -167,10 +162,35 @@ private static Collector<ScopusFile, Object, ArrayList<ScopusFile>> collectRemov
...
+        return !(ELSEVIER_HOST.equals(crossrefLink.getUri().getHost())
```
Comment (joachimjorgensen) [inline scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java:186]: Should this be negated? The method is named "isElsevierPlainTextResource()" and with negating it looks like it checks if it is _not_ an ElsevierTextResource

File: scopus-import/src/test/java/no/sikt/nva/scopus/conversion/ScopusFileConverterTest.java [hunk 3]
```diff
@@ -101,6 +111,82 @@ void shouldNotCreateAssociatedArtifactFromLinkWhichAlsoIsAResourcePrimaryUrl()
...
+        scopusData.getDocument().getMeta().setOpenAccess(randomOpenAccessWithDownloadUrl(firstUrl, secondUrl));
+        mockDownloadUrlResponse();
+
+        var files = fileConverter.fetchAssociatedArtifacts(scopusData.getDocument());
+
+        assertThat(files, hasSize(1));
+    }
+
+    @Test
+    void shouldRemoveFileFromDoiWhenFileIsFromElsveierAndHasPlainTextContentType()
```
Comment (joachimjorgensen) [inline scopus-import/src/test/java/no/sikt/nva/scopus/conversion/ScopusFileConverterTest.java:128]: Typo Elsevier

File: scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java (hunk 2 above)
Comment (truhacevkir) [reply scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java:186]: my bad, it should be isNot... thanks!

File: scopus-import/src/test/resources/crossrefResponseWithElsveierFileToRemove.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (joachimjorgensen) [inline scopus-import/src/test/resources/crossrefResponseWithElsveierFileToRemove.json:1]: Typo in filename Elsevier

**Review by joachimjorgensen — APPROVED** [review]

**Pull request merged**